playbooks/build-heat-stack.yml

```diff
@@ -19,16 +19,18 @@
 
     - name: Wait for the VM to become accessible
       wait_for:
-        host: "{{ heat_stack.stack.outputs | selectattr('output_key', 'equalto', 'floating_ip') | map(attribute='output_value') | first }}"
+        host: "{{ item.output_value }}"
         port: 22
         state: started
         timeout: 300
+      with_items:
+        - "{{ heat_stack.stack.outputs[0] }}"
 
     - name: Add host to the dynamic inventory
       add_host:
         name: "{{ item.name }}"
-        ansible_host: "{{ heat_stack.stack.outputs | selectattr('output_key', 'equalto', 'floating_ip') | map(attribute='output_value') | first }}"
+        ansible_host: "{{ item.outputs[0].output_value }}"
         groups: "nginx_vm"
-        ansible_ssh_user: "ubuntu"
+        ansible_ssh_user: "{{ vm_user_account }}"
```

Comment:
A variable named "vm_user_account" is defined in playbooks/group_vars/all.yml.
Let's use it